bundles/org.openhab.binding.enocean/src/main/resources/ESH-INF/thing/MultiFunctionSmokeDetector.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="enocean"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="multiFunctionSmokeDetector">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>Multi Function Smoke Detector</label>
+		<description>Multi Function Sensor like a Smoke Detector (EEP: F6-05, D2-14)</description>
+
```

Comment:
Is there a reason not to define the channels here?

Reply:
The channels of a thing are defined by its EEP. For example there are three smoke detecting EEPs which have all other additional channels next to the smoke detection channel. If I would add the channels directly to the thing I would have a lot of useless channels for certain kind of smoke detectors or would have to split up the smoke detector thing into three thing types (simple smoke detector, smoke detector with temp sensor, smoke detector with air quality sensor). I found both solutions not really userfriendly.
However I am open for any other suggestion.

Reply:
But I assume there are some shared channels? Otherwise, they should IMHO be different things. I think that we should at least define the shared channels here. Having the optional temperature or air quality sensor dynamic sounds good.

Reply:
Correct, all thing types have at least one common/shared channel. This common channel defines the thing type (rollershutter thing type => rollershutter channel etc).
However dividing channels into dynamic and non dynamic ones would make the channel handling a liitle bit more complicated. For example whenever the user changes the EEP, I just remove all channels and readd them according to the selected EEP. However if there would be "static" channels, I had to ensure that these static channels would still be there after the update. So keeping all channels dynamic and just depended on the selected EEP (and not the thing type) simplyfies the channel handling a lot.

Reply:
Alright, personally I'm not such a fan of having it so dynamic :-)